src/memspace.c

```diff
@@ -426,3 +426,133 @@ umfMemspaceMemtargetRemove(umf_memspace_handle_t hMemspace,
     hMemspace->size--;
     return UMF_RESULT_SUCCESS;
 }
+
+static int MemspaceFilter(umf_memspace_handle_t memspace,
+                          umf_memspace_filter_func_t filter, void *args) {
+
+    if (!memspace || !filter) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+    size_t idx = 0;
+    umf_result_t ret;
+    umf_memtarget_handle_t *nodesToRemove =
+        umf_ba_global_alloc(sizeof(*nodesToRemove) * memspace->size);
+
+    if (!nodesToRemove) {
+        return UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY;
+    }
+
+    for (size_t i = 0; i < memspace->size; i++) {
+        int r = filter(memspace, memspace->nodes[i], args);
+        if (r < 0) {
+            return r;
+        } else if (r == 0) {
```

Comment:
what if r>0? assert?

Reply:
Nothing - we are not doing anything with this memtarget